pinot-controller/src/main/java/com/linkedin/pinot/controller/api/storage/PinotFS.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.api.storage;
+
+import org.apache.commons.configuration.Configuration;
+
+/**
+ * The PinotFS is intended to be a thin wrapper on top of different filesystems. Each storage type that is implemented
+ * will implement this class for filesystem-specific operations
+ */
+public interface PinotFS {
+  /**
+   * Initializes the configurations specific to that filesystem. For instance, any security related parameters can be
+   * initialized here
+   * @param config
```

Comment:
Would be good to add javadocs; if not, might be better to remove the @param/@return/@throws instead of partial docs.